python/pyspark/context.py

```diff
@@ -156,7 +156,7 @@ def _do_init(self, master, appName, sparkHome, pyFiles, environment, batchSize,
         self.sparkHome = self._conf.get("spark.home", None)
 
         # Let YARN know it's a pyspark app, so it distributes needed libraries.
-        if self.master == "yarn-client":
+        if self.master == "yarn":
```

Comment:
Thanks for quick reply.

I think my problem is building the whole project with -Pscala-2.10. I'm not sure if I missed some doc for building spark 2.x with scala-2.10, or spark 2.x is not compatible with scala-2.10?